src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -414,19 +414,21 @@ SearchCachedShardInterval(Datum partitionColumnValue, ShardInterval **shardInter
 bool
 SingleReplicatedTable(Oid relationId)
 {
-	List *shardIntervalList = LoadShardList(relationId);
-	ListCell *shardIntervalCell = NULL;
+	List *shardList = LoadShardList(relationId);
+	List *shardPlacementList = NIL;
+	Oid shardId = INVALID_SHARD_ID;
 
-	foreach(shardIntervalCell, shardIntervalList)
+	if (list_length(shardList) <= 1)
```

Comment:
What is this check for ? reference table ?

Reply:
To prevent crash for append/range partitioned tables without any shards

Reply:
could you add a line describing that ?